File: use_multcpu.py
#!/home/touchiyama/.pyenv/versions/anaconda3-2019.03/bin/python

import sys
import subprocess
from concurrent import futures

args=sys.argv
f_name=args[1];
N_cpu=int(args[2]);

f=open(f_name,"r")
array=[]
for line in f:
  array.append(line.rstrip('\n'))
f.close()

def run(cmd):
  subprocess.call(cmd,shell=True)
  print('%s' % cmd)

future_list=[]
with futures.ThreadPoolExecutor(max_workers=N_cpu) as executer:
  future=[executer.submit(run,cmd) for cmd in array]
  future_list.append(future)
  _ = futures.as_completed(fs=future_list)

print("finished")

File: use_multcpu.pl
#!/usr/bin/perl

use lib ('/home00/kohanada/bin');
use ForkManager;

$file=$ARGV[0];
$N_cpu=$ARGV[1];
open(in, $file);
while(<in>){
        chomp;
        push(@run, $_);
}
close in;

$pm=new Parallel::ForkManager($N_cpu);
for(my $i=0; $i<@run; $i++){
        my $pid=$pm -> start and next;
        `$run[$i]`;
        $pm->finish;
}

